add_me_to_your_PATH/Dockerfile.template

```diff
@@ -43,18 +45,21 @@ RUN apt-get update && apt-get install -y \
     gcc \
     && rm -rf /var/lib/apt/lists/*
 
-RUN julia -e 'using Pkg; Pkg.Registry.add("General")'
-
-# Optionally install a private Registry
-# RUN julia -e 'using Pkg; Pkg.Registry.add(RegistrySpec(name="MyPrivate", url="https://github.com/my-github-org/MyPrivateRegistry.git"))'
+# Install the General registry and optionally a private registry
+ARG PRIVATE_REGISTRY_URL=""
+RUN --mount=type=secret,id=github_token \
+    julia -e 'using Pkg; \
+              !isempty(ENV["PRIVATE_REGISTRY_URL"]) && Pkg.Registry.add(RegistrySpec(url=ENV["PRIVATE_REGISTRY_URL"])); \
+              Pkg.Registry.add("General")'
 
 # Instantiate the Julia project environment
 COPY --from=pinned-extractor *Project.toml *Manifest.toml /JuliaProject/
 
 RUN julia --project=/JuliaProject -e 'using Pkg; Pkg.instantiate(); Pkg.build(); Pkg.precompile()'
 
 COPY sysimage.jl /JuliaProject/sysimage.jl
-RUN julia --project=/JuliaProject -e 'include("/JuliaProject/sysimage.jl")'
+RUN --mount=type=secret,id=github_token \
+    julia --project=/JuliaProject -e 'include("/JuliaProject/sysimage.jl")'
```

Comment:
I've added the `--mount` to avoid having Julia complain it can't update the private registry for `RUN` instructions where this would occur. Note that only in this particular case it shouldn't be required as we're only adding PackageCompiler.jl which is in the public registry